pimcore/models/Asset.php

```
@@ -987,50 +992,55 @@ public function delete()
+                $this->deletePhysicalFile();
+            }
 
-        $versions = $this->getVersions();
-        foreach ($versions as $version) {
-            $version->delete();
-        }
+            $versions = $this->getVersions();
+            foreach ($versions as $version) {
+                $version->delete();
+            }
 
-        // remove permissions
-        $this->getDao()->deleteAllPermissions();
+            // remove permissions
+            $this->getDao()->deleteAllPermissions();
 
-        // remove all properties
-        $this->getDao()->deleteAllProperties();
+            // remove all properties
+            $this->getDao()->deleteAllProperties();
 
-        // remove all metadata
-        $this->getDao()->deleteAllMetadata();
+            // remove all metadata
+            $this->getDao()->deleteAllMetadata();
 
-        // remove all tasks
-        $this->getDao()->deleteAllTasks();
+            // remove all tasks
+            $this->getDao()->deleteAllTasks();
 
-        // remove dependencies
-        $d = $this->getDependencies();
-        $d->cleanAllForElement($this);
+            // remove dependencies
+            $d = $this->getDependencies();
+            $d->cleanAllForElement($this);
 
-        // remove from resource
-        $this->getDao()->delete();
+            // remove from resource
+            $this->getDao()->delete();
 
-        // empty object cache
-        $this->clearDependentCache();
+            // empty asset cache
+            $this->clearDependentCache();
 
-        //set object to registry
-        \Pimcore\Cache\Runtime::set('asset_' . $this->getId(), null);
+            // clear asset from registry
+            \Pimcore\Cache\Runtime::set('asset_' . $this->getId(), null);
+        } catch (\Exception $e) {
+            \Pimcore::getEventDispatcher()->dispatch(AssetEvents::POST_DELETE_FAILURE, new AssetEvent($this));
```

is there a throw $e missing at the end of the exception handler  ?

---

we definitely still should throw the exceptions.

---

Done :-) thanks for the catch 👍